src/megaapi_impl.cpp

```diff
@@ -5157,6 +5157,9 @@ void MegaApiImpl::init(MegaApi *api, const char *appKey, MegaGfxProcessor* proce
     httpio->unlock();
 #endif
 
+    mHeartBeatMonitor = ::mega::make_unique<MegaHeartBeatMonitor>(client);
+    listeners.insert(mHeartBeatMonitor.get());
```

Comment:
I think this listener should be unregistered at some point. Furthermore, IMO it would be better to use the standard method to add it: `MegaApi::addListener()`. Perhaps, it could register itself if we pass `this` instead of `client`, and unregister at dtor :thinking: